core/string/ustring.cpp

```diff
@@ -3206,48 +3188,32 @@ int String::findmk(const Vector<String> &p_keys, int p_from, int *r_key) const {
 		return -1;
 	}
 
-	//int src_len=p_str.length();
 	const String *keys = &p_keys[0];
-	int key_count = p_keys.size();
-	int len = length();
+	const int key_count = p_keys.size();
+	const int len = length();
 
 	if (len == 0) {
-		return -1; // won't find anything!
+		return -1; // Won't find anything!
 	}
 
 	const char32_t *src = get_data();
 
 	for (int i = p_from; i < len; i++) {
-		bool found = true;
 		for (int k = 0; k < key_count; k++) {
-			found = true;
-			if (r_key) {
-				*r_key = k;
-			}
-			const char32_t *cmp = keys[k].get_data();
-			int l = keys[k].length();
+			const int str_len = keys[k].length();
```

Comment:
from a cache point of view, these 2 loops should swap, to not iterate those keys everytime (and potentially kick one of them outised the L1 cache before next iteration) (aka : loop over the key first, then over the length)

Reply:
The function needs to exit early when the first key is found anywhere. If `keys` is the outer loop, then you'll incorrectly iterate the whole string before considering the next key.